src/hyperion/experiment_plans/tests/test_flyscan_xray_centre_plan.py

```diff
@@ -321,11 +309,31 @@ def test_logging_within_plan(
             mock_subscriptions,
         )
     )
+    assert fake_fgs_composite.smargon.stub_offsets.center_at_current_position.get() == 1
 
-    run_gridscan.assert_called_once_with(fake_fgs_composite, test_fgs_params)
-    array_arg = move_xyz.call_args.args[1:4]
-    np.testing.assert_array_almost_equal(array_arg, np.array([0.05, 0.15, 0.25]))
-    move_xyz.assert_called_once()
+
+@patch("hyperion.experiment_plans.flyscan_xray_centre_plan.run_gridscan", autospec=True)
+@patch("hyperion.experiment_plans.flyscan_xray_centre_plan.move_x_y_z", autospec=True)
+def test_given_gridscan_fails_to_centre_then_stub_offsets_not_set(
+    move_xyz: MagicMock,
+    run_gridscan: MagicMock,
+    fake_fgs_composite: FlyScanXRayCentreComposite,
+    test_fgs_params: GridscanInternalParameters,
+    RE: RunEngine,
+):
+    move_xyz.side_effect = Exception()
+    mock_subscriptions = MagicMock()
+    mock_subscriptions.zocalo_handler.wait_for_results.return_value = ((0, 0, 0), None)
+
+    with pytest.raises(Exception):
```

Comment:
I think it would be better to make the side effect of `move_xyz` some _specific_ exception type, and then only assert that one exception is thrown. This will catch (and pass) on all sorts of things currently - including programming errors like misnamed attributes and the like...